src/dbnode/persist/fs/seek_manager.go

```diff
@@ -423,7 +424,20 @@ func (m *seekerManager) newOpenSeeker(
 	// Set the unread buffer to reuse it amongst all seekers.
 	seeker.setUnreadBuffer(m.unreadBuf.value)
 
-	resources := m.getSeekerResources()
+	var (
+		resources = m.getSeekerResources()
+		blm       = m.blockRetrieverOpts.BlockLeaseManager()
+	)
+	_, err = blm.OpenLatestLease(m, block.LeaseDescriptor{
+		Namespace:  m.namespace,
+		Shard:      shard,
+		BlockStart: blockStart,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("err opening latest lease: %v", err)
+	}
+
+	// TODO(rartoul): Pass volume obtained from OpenLatestLease here.
```

Comment:
Could you and justin perhaps use like a TODO_COLDFLUSH(...) or something tag so we can track all these short term things needing updates without forgetting about any one place?  I know not part of your change but on justin's PR there's places in the namespace reader cache where it could be very easy to forget about updating the volume index passed into the reader.

Reply:
Yeah Justin actually has an issue filed for that: https://github.com/m3db/m3/issues/1732

That said I think `TODO_OUT_OF_ORDER_WRITES` (since thats what the issue tag is called) is a good idea. I'll use that here and I just filed this issue to track: #1746